internal/diff/diff.go

```diff
@@ -31,23 +31,22 @@ import (
 	"io/ioutil"
 	"os"
 	"os/exec"
-	"path/filepath"
 	"runtime"
 )
 
 // Do does a diff between an input and output.
 func Do(input []byte, output []byte, filename string) ([]byte, error) {
 	f1, err := writeTempFile("", "prototool-diff", input)
+	defer func() { _ = os.Remove(f1) }()
```

Comment:
Why is this above the error statement? If err is not nil, then f1 could be nil, meaning you're calling `os.Remove` on nil. This should probably be put back to where it was, same below.

Reply:
The `f1` type is a string, so the default value will be `""`.
It's above so that we always remove the file (even in the presence of an error. This was necessary for errors in calls to `*os.File.Close` where the file was non-nil.